pinecone/client.go

```diff
@@ -16,25 +18,41 @@ import (
 
 type Client struct {
 	apiKey     string
+	headers    map[string]string
 	restClient *control.Client
 	sourceTag  string
-	headers    map[string]string
+}
+
+type NewClientBaseParams struct {
+	Headers    map[string]string
+	Host       string
+	RestClient *http.Client
+	SourceTag  string
 }
 
 type NewClientParams struct {
-	ApiKey     string            // required unless Authorization header provided
-	SourceTag  string            // optional
-	Headers    map[string]string // optional
-	RestClient *http.Client      // optional
+	ApiKey     string
+	Headers    map[string]string
+	Host       string
+	RestClient *http.Client
+	SourceTag  string
 }
 
 func NewClient(in NewClientParams) (*Client, error) {
-	clientOptions, err := buildClientOptions(in)
+	clientOptions, err := in.buildClientOptions()
 	if err != nil {
 		return nil, err
 	}
 
-	client, err := control.NewClient("https://api.pinecone.io", clientOptions...)
+	controlHostOverride := valueOrFallback(in.Host, os.Getenv("PINECONE_CONTROLLER_HOST"))
```

Comment:
"CONTROL" vs "CONTROLLER"? (Assuming this is talking about the control plane?

Reply:
I went with `CONTROLLER` because that's the env key that's used in the Python client for the same config: https://github.com/pinecone-io/pinecone-python-client/blob/1fad23f00c114aa8ef33242b91b969da2825c360/pinecone/config/pinecone_config.py#L15

I'm open to changing it up if we want.